cid.go

```diff
@@ -37,10 +37,32 @@ import (
 // UnsupportedVersionString just holds an error message
 const UnsupportedVersionString = "<unsupported cid version>"
 
+// ErrInvalidCid is an error that indicates that a CID is invalid.
+type ErrInvalidCid struct {
+	Err error
+}
+
+func (e *ErrInvalidCid) Error() string {
+	return fmt.Sprintf("invalid cid: %s", e.Err)
+}
+
+func (e *ErrInvalidCid) Unwrap() error {
+	return e.Err
+}
+
+func (e *ErrInvalidCid) Is(err error) bool {
+	switch err.(type) {
+	case *ErrInvalidCid:
```

Comment:
if you want to be sure then check for both the pointer and non-pointer form here